std/experimental/allocator/gc_allocator.d

```diff
@@ -80,6 +80,25 @@ struct GCAllocator
         return true;
     }
 
+    /// Ditto
+    size_t goodAllocSize(size_t n) shared
+    {
+        if(n == 0)
+            return 0;
+        import core.bitop: bsr;
+
+        auto largestBit = bsr(n);
```

Comment:
auto largestBit = bsr(n - 1) + 1;

This is wrong for 0, ut you already bailed out zeros, and you don't need to special case power of 2.